### PR TITLE
fix for #395 It's impossible to parse CSV file with more than 512 col…

### PR DESCRIPTION
**Pull request by Ebalaitung**
Thanks for contributing.

- [X ] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](https://developercertificate.org)

## Description

Added setting for maximal columns in CVS file in CsvReaderOption

## Testing

Added positive and negative tests for CsvReader for 1000 columns csv file.

### CODE REVIEW

File: core/src/main/java/tech/tablesaw/io/csv/CsvReader.java [hunk 1]
```diff
@@ -578,6 +578,9 @@ private ColumnType selectType(List<ColumnType> typeCandidates) {
     private CsvParser csvParser(CsvReadOptions options) {
         CsvParserSettings settings = new CsvParserSettings();
         settings.setFormat(csvFormat(options));
+        if(options.maxNumberOfColumns() != null){
```
Comment (benmccann) [inline core/src/main/java/tech/tablesaw/io/csv/CsvReader.java]: Can you please add a space after the `if` and before the `{`?

Comment (Ebalaitung) [reply core/src/main/java/tech/tablesaw/io/csv/CsvReader.java]: fixed

File: core/src/main/java/tech/tablesaw/io/csv/CsvReadOptions.java [hunk 2]
```diff
@@ -264,6 +271,15 @@ public Builder columnTypes(ColumnType[] columnTypes) {
             return this;
         }
 
+        /**
+         * Defines maximal value of columns in csv file.
+         * @param maxNumberOfColumns - must be positive integer. Default is 512.         *
+         */
+        public Builder maxNumberOfColumns(Integer maxNumberOfColumns){
```
Comment (benmccann) [inline core/src/main/java/tech/tablesaw/io/csv/CsvReadOptions.java]: can you add a space before the `{`

Comment (Ebalaitung) [reply core/src/main/java/tech/tablesaw/io/csv/CsvReadOptions.java]: fixed

**Comment by lwhite1** [comment]
thanks, Ben, for taking care of this.

**Comment by benmccann** [comment]
I wonder if we should change the default value of this new setting to 10000 so that users will be unlikely to hit the limit. That will use only 40kb of memory (https://github.com/uniVocity/univocity-parsers/issues/286#issuecomment-434929433) and will make Tablesaw easier to use out of the box

**Comment by Ebalaitung** [comment]
@benmccann sounds reasonable, I guess it'll cover 95% use cases. If everybody  are aligned with this approach, I can create PR.

**Comment by benmccann** [comment]
I agree. Unfortunately, univocity-parser doesn't. If you want to change it here I'm supportive of that

**Pull request merged**